Sources/PubNubChatComponents/Message List/MessageListViewController.swift

```diff
@@ -133,4 +209,162 @@ open class ChatViewController: CollectionViewComponent {
   fileprivate var isUserInitiatedScrolling: Bool {
     return collectionView.isDragging || collectionView.isDecelerating
   }
+  
+  // MARK: UIViewControllerTransitioningDelegate
+
+  @objc(presentationControllerForPresentedViewController:presentingViewController:sourceViewController:)
+  public func presentationController(
+    forPresented presented: UIViewController,
+    presenting: UIViewController?,
+    source: UIViewController
+  ) -> UIPresentationController? {
+    
+    return AddMessageReaction.PresentationController(
+      presentedViewController: presented,
+      presentingViewController: presenting,
+      messageCell: gestureRecognizedCell!,
+      messageReactionsComponentSize: preferredPickerViewSize
+    )
+  }
+  
+  @objc(animationControllerForDismissedController:)
+  public func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return AddMessageReaction.AnimationController(duration: 0.15, isBeingPresented: false)
+  }
+  
+  @objc(animationControllerForPresentedController:presentingController:sourceController:)
+  public func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return AddMessageReaction.AnimationController(duration: 0.25, isBeingPresented: true)
+  }
+}
+
+// MARK: Showing message reactions
+
+private enum AddMessageReaction {
+  
+  // MARK: Custom presentation controller (UIPresentationController)
+
+  @objc
+  class PresentationController: UIPresentationController {
+    private let messageCell: MessageListItemCell?
+    private let messageReactionsComponentSize: CGSize
+    
+    private lazy var dimmingView: UIView = {
+      let view = UIView()
+      
+      view.frame = containerView?.bounds ?? .zero
+      view.backgroundColor = .clear
+      view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(onTapGestureRecognized(_:))))
```

Comment:
doesn't superview has property with the same name? if so it can be confusing

Reply:
This code is in the scope of a class that inherits from `UIPresentationController`. And the `UIPresentationController` class has no `UIView` as an ancestor.

Reply:
changes are not needed if ancestor doesnt have it, thanks for explanation